### PR TITLE
[Lambda] Support Drift Detection for Lambda

### LINKED ISSUE #5182: [Lambda] Support Drift Detection for Lambda

### PR DESCRIPTION
**Pull request by t-kikuc**
**What this PR does / why we need it**:

Supported DriftDetection for Lambda

When OutOfSync:
    image

Restrictions: 
- The following fields in [function.yaml](https://pipecd.dev/docs-v0.48.x/user-guide/configuration-reference/#specific-functionyaml) are ignored because livestates don't have them.
  - `source`
  - `s3Bucket`
  - `s3Key`
  - `s3ObjectVersion`

**Which issue(s) this PR fixes**:

Fixes #5182

**Does this PR introduce a user-facing change?**: Users can see drifts.

- **How are users affected by this change**: N/A
- **Is this breaking change**: no
- **How to migrate (if breaking change)**: no

**Note:**

To use this feature, users need to add the following IAM Actions to their Piped's IAM Role.

- [GetFunction](https://docs.aws.amazon.com/lambda/latest/dg/API_GetFunction.html)
- [ListFunctions](https://docs.aws.amazon.com/lambda/latest/dg/API_ListFunctions.html)

Later I'll describe the details in another PR.

### CODE REVIEW

**Review by ffjlabo — DISMISSED** [review]
LGTM

File: pkg/app/piped/driftdetector/lambda/detector.go [hunk 1]
```diff
@@ -13,3 +13,385 @@
...
+		keys = append(keys, k)
+	}
+	slices.Sort(keys)
+
+	sorted := make(map[string]string)
+	for _, k := range keys {
+		sorted[k] = m[k]
+	}
+	return sorted
+}
```
Comment (Warashi) [inline pkg/app/piped/driftdetector/lambda/detector.go:276]: The for-range for map[string]string is always looped in random order.
So, this code may not work as expected.

**Review by Warashi — COMMENTED** [review]
I commented about sorting maps.
I think the code I commented on is not needed because it does not work as expected, but the test you wrote is passed.

File: pkg/app/piped/driftdetector/lambda/detector.go (hunk 1 above)
Comment (t-kikuc) [reply pkg/app/piped/driftdetector/lambda/detector.go:276]: I understood, what a meaningless code... 😅 
let me re-think...

Comment (t-kikuc) [reply pkg/app/piped/driftdetector/lambda/detector.go]: @Warashi 
Thank you, I removed the unnecessary sort.  [be00911](https://github.com/pipe-cd/pipecd/pull/5186/commits/be009116934f76d32d349bc5258eb5f48d701440)

Some slices need to be sorted, but maps do NOT need to be sorted.

**Review by Warashi — APPROVED** [review]
LGTM

**Comment by t-kikuc** [comment]
@ffjlabo Would you please review again?

This is the only one change: https://github.com/pipe-cd/pipecd/pull/5186#discussion_r1745009233

**Comment by ffjlabo** [comment]
Sorry for the rough review🙏
I rechecked the new fix. It is fine.

**Review by ffjlabo — APPROVED** [review]
🚀

**Pull request merged**